spotbugs/src/main/java/edu/umd/cs/findbugs/detect/AtomicOperationsCombinedDetector.java

```diff
@@ -0,0 +1,323 @@
+            if (invokedField != null) {
+                bugInstance.addField(invokedField);
+            } else if (localVariableAnnotation != null) {
+                bugInstance.add(localVariableAnnotation);
+            }
+            if (invokedMethod != null) {
+                bugInstance.addCalledMethod(invokedMethod);
+            }
+            return bugInstance;
+        }
+    }
+
+    private final BugReporter bugReporter;
+    private final BugAccumulator bugAccumulator;
+
+    private final Set<XField> interestingFields = new HashSet<>();
+    private final Set<XField> combinedAtomicFields = new HashSet<>();
+
+    private final Map<Method, Map<XField, List<BugPrototype>>> interestingFieldCalls = new HashMap<>();
+    private final Map<Method, List<BugPrototype>> interestingLocalVariableCalls = new HashMap<>();
+
+    private final Set<XMethod> unsynchronizedPrivateMethods = new HashSet<>();
+
+    public AtomicOperationsCombinedDetector(BugReporter bugReporter) {
+        this.bugReporter = bugReporter;
+        this.bugAccumulator = new BugAccumulator(bugReporter);
+    }
+
+    @Override
+    public void visitClassContext(ClassContext classContext) {
+        try {
+            for (Method method : classContext.getMethodsInCallOrder()) {
+                collectInterestingFields(classContext, method);
+            }
+            for (Method method : classContext.getMethodsInCallOrder()) {
+                analyzeInterestingFields(classContext, method);
+            }
+            removePresynchronizedPrivateMethodCalls();
+            accumulateInterestingFields();
+            accumulateLocalVariables();
+        } catch (CheckedAnalysisException e) {
+            bugReporter.logError(String.format("Detector %s caught exception while analyzing class %s",
+                    getClass().getName(), classContext.getJavaClass().getClassName()), e);
+        } finally {
+            clearProperties();
+        }
+    }
+
+    private void collectInterestingFields(ClassContext classContext, Method method) throws CFGBuilderException {
+        CFG cfg = classContext.getCFG(method);
+        ConstantPoolGen cpg = classContext.getConstantPoolGen();
+
+        for (Location location : cfg.orderedLocations()) {
+            InstructionHandle handle = location.getHandle();
+            Instruction instruction = handle.getInstruction();
+
+            if (instruction instanceof PUTFIELD) {
+                OpcodeStack stack = OpcodeStackScanner.getStackAt(classContext.getJavaClass(), method, handle.getPosition());
+                OpcodeStack.Item stackItem = stack.getStackItem(0);
+                if (isInterestingField(stackItem.getReturnValueOf())) {
+                    interestingFields.add(XFactory.createXField((FieldInstruction) instruction, cpg));
+                }
+            }
+        }
+    }
+
+    private static boolean isInterestingField(ClassMember classMember) {
```

Comment:
As far as I can see this method is only called once with an `XMethod`. Is there any reason why the parameter is not an `XMethod`?

Reply:
I removed the "interesting" word from variables and methods.

Reply:
Fixed the type, thanks!